packages/@sanity/form-builder/src/inputs/ObjectInput/fieldGroups/Group.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {forwardRef, useEffect} from 'react'
+import {Tab} from '@sanity/ui'
+import {FieldGroup} from '@sanity/types/src'
+import {unstable_useConditionalProperty as useConditionalProperty} from '@sanity/base/hooks'
+
+interface GroupType extends FieldGroup {
+  onClick: (string) => void
+  autoFocus?: boolean
+  selected: boolean
+  onStateChange?: (name: string, hidden: boolean) => void
+  parent: unknown
+  'aria-controls': string
+  disabled?: boolean
+}
+
+export const GroupTab = forwardRef(function GroupTab(
+  props: GroupType,
+  ref: React.Ref<HTMLButtonElement>
+) {
+  const {name, title} = props
+  // Separate props for resolving conditional hidden groups
+  const {onClick, onStateChange, parent, hidden, ...group} = props
+
+  const isHidden = useConditionalProperty({
+    checkProperty: hidden,
+    checkPropertyKey: 'hidden',
+    value: group,
+    parent,
+  })
+
+  // Here goes the content of our component
+  const handleClick = React.useCallback(() => {
+    onClick(name)
+  }, [name, onClick])
+
+  useEffect(() => {
+    onStateChange(name, Boolean(isHidden))
+  }, [isHidden, name, onStateChange])
+
+  if (isHidden) {
+    return null
+  }
+
+  return (
+    <Tab
+      data-testid={`group-tab-${name}`}
+      size={1}
+      id={`${name}-tab`}
+      label={title || name}
+      title={title || name}
+      onClick={handleClick}
+      {...group}
+      ref={ref}
+    />
+  )
+})
+
+export const GroupOption = (props: Omit<GroupType, 'onClick' | 'autofocus'>) => {
+  const {name, title, ...rest} = props
+  // Separate props for resolving conditional hidden groups
+  const {autoFocus, selected, hidden, parent, onStateChange, ...group} = props
+  const isHidden = useConditionalProperty({
+    checkProperty: hidden,
+    checkPropertyKey: 'hidden',
+    value: group,
+    parent,
+  })
+
+  useEffect(() => {
+    onStateChange(name, Boolean(isHidden))
```

Comment:
Same concern as above re: triggering an update in the parent component as a side effect of a state change in this. Would hope we can lift this calculation up to a common ancestor so the required data can be shared between child components.